tensorflow/python/keras/metrics.py

```diff
@@ -3342,6 +3342,9 @@ def categorical_accuracy(y_true, y_pred):
   Returns:
     Categorical accuracy values.
   """
+  #assert if both have same shape
+  assert y_true.shape==y_pred.shape,"Both inputs must have same shapes and must be one-hot encoded. Use tf.one_hot() to one hot encode the inputs."
```

Comment:
y_true.shape returns the static shape, please replace with tf.shape(y_true), as tf.shape(..) resolves the actual shape at runtime.

Reply:
I'll do that 👍